src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/CpsPackageReferenceProjectProvider.cs

```diff
@@ -56,15 +56,32 @@ public bool TryCreateNuGetProject(EnvDTE.Project dteProject, ProjectSystemProvid
             {
                 return false;
             }
+            var restoreProjectStyle = string.Empty;
+            var targetFramework = string.Empty;
+            var targetFrameworks = string.Empty;
 
-            // check for RestoreProjectStyle property and if set to PackageReference and project also has CPS 
-            // capability then only mark it as CPS based project.
-            if (!string.IsNullOrEmpty(context.NuGetProjectStyle) &&
-                !context.NuGetProjectStyle.Equals(ProjectStyle.PackageReference.ToString(), StringComparison.OrdinalIgnoreCase))
+            var hasRestoreProjectStyle = context
+                .MSBuildProperties
+                .TryGetValue(ProjectSystemProviderContext.RESTORE_PROJECT_STYLE, out restoreProjectStyle) 
+                && !string.IsNullOrEmpty(restoreProjectStyle);
+
+            var hasTargetFramework = context
+                .MSBuildProperties
+                .TryGetValue(ProjectSystemProviderContext.TARGET_FRAMEWORK, out targetFramework)
+                && !string.IsNullOrEmpty(restoreProjectStyle);
+
+            var hasTargetFrameworks = context
+                .MSBuildProperties
+                .TryGetValue(ProjectSystemProviderContext.TARGET_FRAMEWORKS, out targetFrameworks)
+                && !string.IsNullOrEmpty(restoreProjectStyle);
+
+            // check for RestoreProjectStyle property is set and if set to PackageReference then return false
+            if (hasRestoreProjectStyle && !restoreProjectStyle.Equals(ProjectStyle.PackageReference.ToString(), StringComparison.OrdinalIgnoreCase))
             {
                 return false;
             }
-            else if (!hierarchy.IsCapabilityMatch("CPS"))
+            // Check if the project is not CPS capable or if it is CPS capable then it does not have TargetFramework(s), if so then return false
+            else if (!hierarchy.IsCapabilityMatch("CPS") || !(hasTargetFramework || hasTargetFrameworks))
```

Comment:
nit: the CPS check could be done at the start to immediately return false if it isn't a CPS project. It could speed things up for the primary scenario where this gets hit all the time for non-CPS projects.